src/Symfony/Component/Notifier/Bridge/FakeSms/FakeSmsTransport.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\FakeSms;
+
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author James Hemery <james@yieldstudio.fr>
+ */
+final class FakeSmsMailTransport extends AbstractTransport
+{
+    private $mailer;
+    private $to;
+    private $from;
+
+    public function __construct(MailerInterface $mailer, string $to, string $from, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        parent::__construct($client, $dispatcher);
+
+        $this->mailer = $mailer;
+        $this->to = $to;
+        $this->from = $from;
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('fakesms+mail://%s?to=%s&from=%s', $this->getEndpoint(), $this->to, $this->from);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    /**
+     * @param MessageInterface|SmsMessage $message
+     *
+     * @throws TransportExceptionInterface
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$this->supports($message)) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $email = (new Email())
+            ->from($this->from)
+            ->to($this->to)
+            ->subject('New SMS on '.$message->getPhone())
+            ->html($message->getSubject())
+            ->text($message->getSubject());
+
+        $this->mailer->send($email);
+
+        return new SentMessage($message, (string) $this);
```

Comment:
Shall we do some error handling here?

Reply:
@OskarStark Yes I think it is better, however the Notifier TransportException requires a Response instance, maybe make it nullable?

Reply:
As its meant to be a dev notifier its not so important IMHO, I am fine as is for a first version.